network/network.go

```diff
@@ -187,6 +187,13 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 
 	var wg sync.WaitGroup
 
+	pods, err := c.k8sClient.CoreV1().Pods(c.namespace).List(metav1.ListOptions{})
```

Comment:
I don't think this will necessarily solve the issue - isn't the issue that the loop below takes too long and then the pods diverge from the service?

I would therefor try to find pods in every loop iteration maybe? Not sure though.

Reply:
I thought these goroutines are all triggered at basically the same time and then return individually to the wait group. I thought the timing issue was that between scrapes a Pod got removed but somehow the host remained to be dialed. Maybe @giantswarm/team-magic can shed some light

Reply:
A previous commit checked the Pod existence per-host in the loop and I could go back to that if it's necessary. Was just hoping to avoid unnecessary api calls

Reply:
Yeah, I don't see how this helps - we're fetching all the endpoints every collect anyway.

I don't see how there's anyway to check pods exist before pinging them, as they could die immediately after the check - there's a bit of a race condition here. Checking if we _just_ pinged a dead pod in error handling could be a way forward.

Reply:
Actually I like that better because then we only need an API call when something failed anyway